core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -309,7 +309,7 @@ protected DeterministicKeyChain(DeterministicSeed seed) {
      * this method to watch an arbitrary fragment of some other tree, this limitation may be removed in future.
      */
     public DeterministicKeyChain(DeterministicKey watchingKey, long creationTimeSeconds) {
-        checkArgument(watchingKey.isPubKeyOnly(), "Private subtrees not currently supported");
```

Comment:
You're probably right, but then we need to somehow change the failing tests. Because what they currently seem to do is creating a private subtree.

I reverted this one change and rebased on master.